core/src/main/scala/org/http4s/parser/ScalazDeliverySchemes.scala

```diff
@@ -14,10 +14,10 @@ private[http4s] object ScalazDeliverySchemes {
   // scalastyle:off public.methods.have.type
   implicit def Disjunction[L <: HList, Out](implicit unpack: Unpack.Aux[L, Out]) =
     new DeliveryScheme[L] {
-      type Result = ParseFailure \/ Out
+      type Result = ParseError \/ Out
       def success(result: L) = \/-(unpack(result))
-      def parseError(error: ParseError) = -\/(ParseFailure("", errorFormattter.formatExpectedAsString(error)))
-      def failure(error: Throwable) = -\/(ParseFailure("Exception during parsing.", error.getMessage))
+      def parseError(error: ParseError) = -\/(error)
+      def failure(error: Throwable) = throw error
```

Comment:
I haven't looked up where parboiled will call failure, but presumably this would be a case of throwing from inside a combinator or assert failure, but I agree: if its not a ParseFailure its an exceptional situation and throwing seems appropriate.